ldms/src/sampler/spank/slurm_notifier.c

```diff
@@ -612,7 +682,49 @@ jbuf_t make_exit_data(spank_t sh, const char *event, const char *context)
 	return jb;
 }
 
-jbuf_t make_task_init_data(spank_t sh, const char *event, const char *context)
+jbuf_t make_step_init_data(spank_t sh, const char *event)
+{
+	jbuf_t jb;
+
+	jb = jbuf_new(); if (!jb) goto out_1;
+	jb = jbuf_append_str(jb, "{"); if (!jb) goto out_1;
+	jb = jbuf_append_attr(jb, "schema", "\"slurm_job_step_data\","); if (!jb) goto out_1;
+	jb = jbuf_append_attr(jb, "event", "\"%s\",", event); if (!jb) goto out_1;
+	jb = jbuf_append_attr(jb, "timestamp", "%d,", time(NULL)); if (!jb) goto out_1;
+	jb = jbuf_append_attr(jb, "context", "\"%s\",", context_str(sh)); if (!jb) goto out_1;
+	jb = jbuf_append_attr(jb, "data", "{"); if (!jb) goto out_1;
+	jb = _append_item_u32(sh, jb, "job_id", S_JOB_ID, ','); if (!jb) goto out_1;
+	jb = _append_item_u32(sh, jb, "nodeid", S_JOB_NODEID, ','); if (!jb) goto out_1;
+	jb = _append_item_u32(sh, jb, "step_id", S_JOB_STEPID, ','); if (!jb) goto out_1;
+	jb = _append_item_u64(sh, jb, "alloc_mb", S_STEP_ALLOC_MEM, ','); if (!jb) goto out_1;
+	jb = _append_item_u16(sh, jb, "ncpus", S_JOB_NCPUS, ','); if (!jb) goto out_1;
+	jb = _append_item_u32(sh, jb, "nnodes", S_JOB_NNODES, ','); if (!jb) goto out_1;
+	jb = _append_item_u32(sh, jb, "local_tasks", S_JOB_LOCAL_TASK_COUNT, ','); if (!jb) goto out_1;
+	jb = _append_item_u32(sh, jb, "total_tasks", S_JOB_TOTAL_TASK_COUNT, ' '); if (!jb) goto out_1;
+	jb = jbuf_append_str(jb, "}}");
+ out_1:
+	return jb;
+}
+
+jbuf_t make_step_exit_data(spank_t sh, const char *event)
```

Comment:
It is not clear why this function should have an "event" parameter when the only place this can possibly be called is at "step_exit".